src/browser/page-switch.js

```diff
@@ -31,21 +30,35 @@ define((require, exports, moudle) => {
         }
       }
     }, [
-      DOM.span({
+      DOM.div({
         key: 'thumbnail',
         className: 'tab-thumbnail',
-        style: {backgroundImage: readThumbnailURI(webViewerCursor.get('location'))},
-      }),
+      }, [
+        DOM.img({
+          key: 'image',
+          src: thumbnail,
+          alt: '',
+          onLoad: event => URL.revokeObjectURL(event.target.src)
```

Comment:
> you store thumbnail in the state. That will break on restart, as the blobs are not stored across sessions

I don' really store blobs I store blob uris (see https://github.com/mozilla/browser.html/pull/220/files#diff-ac64a61ab532164b9e3f853bfb7ee2d7R201), but yeah they won't work across the sessions. I think for that we need to implement / revive proper database like pouchdb. But I think we were going to defer that for now.

> you revoke the URL after load, making re-using the blob impossible even in the same session

Sure but we do not re-render those images ever with in the same session, if tab is gone then all associated data is gone anyway. I could store actual blob in the state and create url during render and revoke after it loads but I don't find it to be any better.